app/Core/Ldap/Client.php

```diff
@@ -88,7 +88,12 @@ public function open($server, $port = LDAP_PORT, $tls = LDAP_START_TLS, $verify
             putenv('LDAPTLS_REQCERT=never');
         }
 
-        $this->ldap = @ldap_connect($server, $port);
+        if (is_integer($port)) {
```

Comment:
The default value of `$port` is `389`, so it will always use the deprecated from of `ldap_connect()` unless `$LDAP_PORT` is set to a non-integer value. 

https://github.com/kanboard/kanboard/blob/master/app/constants.php#L66

What about testing if the `$LDAP_SERVER` value starts with `ldap[s]://` instead?